src/platform/android/java/chip/platform/NsdServiceFinderAndResolver.java

```diff
@@ -132,7 +132,12 @@ public void onResolveFailed(NsdServiceInfo serviceInfo, int errorCode) {
                 "Failed to resolve service '" + serviceInfo.getServiceName() + "': " + errorCode);
             chipMdnsCallback.handleServiceResolve(
                 serviceInfo.getServiceName(),
-                serviceInfo.getServiceType(),
+                // Use the target service info since the resolved service info sometimes appends a
```

Comment:
`sometimes appends` sounds like we are not sure what is going on ... any chance we can actually figure out what is going wrong or at least create a github issue for this?

Reply:
This issue is judged not to be an issue of Matter SDK but an issue of Android OS. In certain cases, the serviceType is returned with a ".", such as "._matter._tcp", and in some cases, it is returned without a "." The phrase "sometimes apps" was added to show this in the comment.